INT v2.1 final version

mickeyspiegel opened the pull request:
- Lots of minor editorial and formatting cleanup
- Changed 'switch id' terminology to 'node id'
- Changed 'switch' terminology to 'node' where appropriate
- VXLAN-GPE codepoint changed from 0x80 to 0x82
- Minor rewording and streamlining of "Domain Specific ID" and
  "DS Instruction" definitions.
- In INT-MX header, removed "Checksum Complement" from "Instruction Bitmap",
  since there is no corresponding field in the telemetry report, and that bit position
  is reclaimed for "drop reason".

rsivakolundu commented on telemetry/specs/INT.mdk:
```diff
@@ -209,261 +209,275 @@ of Little Minions paper [^Minions].
+    specification defines the behavior and packet header formats for
+    interoperability between INT devices from different vendors in an INT
+    domain. The INT devices within the same domain must be configured in a
+    onsistent way to ensure interoperability between the devices. Operators of
```
typo - s/onsistent/consistent

mickeyspiegel replied:
Done

rsivakolundu commented on telemetry/specs/INT.mdk:
````diff
@@ -933,52 +956,52 @@ by most software virtual switches and hardware network elements. The VXLAN
-   |                VXLAN Network Identifier (VNI) |   Reserved    |
-   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-`
+```
````
Not clear what three ``` indicates. Seems to be different than one `.

mickeyspiegel replied:
With just `, all the spaces at the beginning of each line were ignored, resulting in the bit numbers being offset by one from where we wanted them.
Changing to ``` fixed that, and also indented the whole thing by however many spaces there are.

mickeyspiegel replied:
If you look at the pdf output you will see the difference

rsivakolundu commented on telemetry/specs/INT.mdk:
````diff
@@ -933,52 +956,52 @@ by most software virtual switches and hardware network elements. The VXLAN
-* 0x80 to 0xFF: Unassigned (shim headers)
-* 0x80: In-band Network Telemetry Header (This value has not been reserved by
-VXLAN GPE specification yet, and is hence subject to change)
+```
````
Same as the earlier comment. Thee ```

rsivakolundu commented on telemetry/specs/INT.mdk:
````diff
@@ -933,52 +956,52 @@ by most software virtual switches and hardware network elements. The VXLAN
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                VXLAN Network Identifier (VNI) |   Reserved    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
````
Same as before. Three ```

rsivakolundu commented on telemetry/specs/INT.mdk:
````diff
@@ -933,52 +956,52 @@ by most software virtual switches and hardware network elements. The VXLAN
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                VXLAN Network Identifier (VNI) |   Reserved    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
````
Same comment as above. Three ```

rsivakolundu commented:
For VXLAN-GPE, the codepoint of 0x82 is being assigned for INT. A similar experimental range is also be defined for VXLAN-GPE. Not clear which way all these codepoint assignments will go.

mickeyspiegel commented:
@rsivakolundu 
I changed the Geneve codepoints to 0x0103 an hour ago.

I changed the VXLAN-GPE codepoint to 0x82 a while ago, but it looks like I missed the examples, will take care of that now.

jklr commented on telemetry/specs/INT.mdk:
```diff
@@ -1309,7 +1326,7 @@ INT-MX Header:
 * The INT-MX header is 12 bytes long. Each metadata requested in the
   INT-MX Header instruction is either 4 bytes or 8 bytes in length. Each INT
-  capable device in the forwarding path will send the requested metadata to the
+  node in the forwarding path will send the requested metadata to the
```
Below we have a link to TR spec 2.0 PDF. It makes sense for INT 2.1 spec.

In the master version of .mdk, we'd want to link to TR.mdk, not a particular PDF. There must be a way to define a variable in mdk, so that we can control INT version and matching TR version in one place.

jklr commented on telemetry/specs/INT.mdk:
```diff
@@ -1335,7 +1352,7 @@ be accessed at the link below:\
     - The largest value of *Hop ML* for baseline and domain specific
       metadata is 31.
-    - An INT-capable device may be limited in the maximum number
+    - An INT node may be limited in the maximum number
```
Do we consider the case where a node is not capable of processing a specific instruction, while it's not limited by the max number of instructions or metadata?

rsivakolundu replied:
I think so. Line 1359-1366 states

An INT hop that cannot process all instructions has the following alternatives:
      - Send to the monitoring system *Hop ML* \* 4 bytes, with all-ones reserved
        value (4 or 8 bytes of 0xFF depending on the length of metadata) for the
        metadata corresponding to instructions it cannot process, or
      - Send to the monitoring system the metadata it can process, update the
        *RepMdBits*, *DSMdBits* and *MD Length* fields appropriately in the
        Telemetry Report.

mickeyspiegel commented:
@jklr 
I could leave the first occurrence in the MX description and delete the other 3.

@rsivakolundu 
Is that ok with you?

rsivakolundu commented:
It is ok with me.

Thanks

-Ramesh
On Wed, May 13, 2020 at 3:04 PM Mickey Spiegel <notifications@github.com>
wrote:

> @jklr <https://github.com/jklr>
> I could leave the first occurrence in the MX description and delete the
> other 3.
>
> @rsivakolundu <https://github.com/rsivakolundu>
> Is that ok with you?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/p4lang/p4-applications/pull/80#issuecomment-628269934>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADZZHFXASZXTT3CIG2E46LTRRMKQRANCNFSM4M2CABQQ>
> .
>

mickeyspiegel commented:
@jklr @rsivakolundu 
I changed all the http references to footnotes, which makes it easier to throw it in multiple places. The redundant sentences "Details ... can be accessed at" have been removed, leaving only one each for metadata semantics YANG and Telemetry Report 2.0

rsivakolundu commented:
I am good with those changes.

-Ramesh

On Wed, May 13, 2020 at 7:30 PM Mickey Spiegel <notifications@github.com>
wrote:

> @jklr <https://github.com/jklr> @rsivakolundu
> <https://github.com/rsivakolundu>
> I changed all the http references to footnotes, which makes it easier to
> throw it in multiple places. The redundant sentences "Details ... can be
> accessed at" have been removed, leaving only one each for metadata
> semantics YANG and Telemetry Report 2.0
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/p4lang/p4-applications/pull/80#issuecomment-628348394>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADZZHFRE2NI5DCI7TEYFDLLRRNJTVANCNFSM4M2CABQQ>
> .
>

jklr commented:
+1
